src/irc/core/irc-session.c

```diff
@@ -67,7 +67,7 @@ static void sig_session_save_server(IRC_SERVER_REC *server, CONFIG_REC *config,
 	config_node_set_str(config, node, "away_reason", server->away_reason);
 	config_node_set_bool(config, node, "emode_known", server->emode_known);
 
-	config_node_set_int(config, node, "sasl_mechanism", server->connrec->sasl_mechanism);
+	config_node_set_str(config, node, "sasl_mechanism", server->connrec->sasl_mechanism);
```

Comment:
We should provide some transparent way to handle this change (int -> string) or use a new field.